### ISSUE TITLE
[Maintenance Windows] Skip module upgrades during maintenance window

### ISSUE THREAD

**Issue by lindnerby** [kyma-project/lifecycle-manager#2101]
Depends on: https://github.com/kyma-project/lifecycle-manager/issues/2163

Once KLM can read maintenance window policies and have them cached, it needs to respect these for modules requiring downtime. Overall, the following cases need to be covered:

- [x] `Module.RequiresDowntime: FALSE`, `Kyma.UseMaintenanceWindows: FALSE` => module updated immediately
- [x] `Module.RequiresDowntime: TRUE`, `Kyma.UseMaintenanceWindows: FALSE` => module updated immediately
- [x] `Module.RequiresDowntime: TRUE`, `Kyma.UseMaintenanceWindows: TRUE` => module updated within next window, in the meantime reconciled in its currently installed version
- [x] `Module.RequiresDowntime: FALSE`, `Kyma.UseMaintenanceWindows: TRUE` => module updated immediately

### AC

- [x] Implemented logic for above cases
- [x] logic unit tested
- [ ] Cases E2E tested, see concept evaluation here: https://github.com/kyma-project/lifecycle-manager/issues/2164

### Related issue
* https://github.com/kyma-project/kyma/issues/18611

**Comment by c-pius** [comment]
I was wondering if it is possible to split this into multiple issues:

- first ticket to make KLM aware of the maintenance windows config (volume mount, kustomize patch, ...)
- second ticket to investigate how we can do the testing (e.g., for e2e testing we need to find out about the granularity of the maintenance windows that can be maintained (minute level), and how to properly configure a test environment to test with it (can we predict the system time of the executed tests or do we need to temper with the system clock, ...))
- third ticket to implement the actual maintenance window logic and testing

Regarding the feature flag, I am wondering if this is needed. Existing E2E tests don't use module templates with `requiresDowntime`, so nothing changes for those, even when maintenance windows are maintained. For the new logic and new tests, starting point will be reading a file from the system. If the file is not found, we just treat it as no policies maintained instead of erroring out. Once we roll this out to production, we can change the latter part to requiring the file and erroring out if not found.

**Comment by nesmabadr** [comment]
For the E2E test, we can add the maintenance windows component to the watcher_local_test Kustomization profile by using the command `kustomize edit add component ../maintenance_windows` from the watcher_local_test path in the workflow

**Comment by c-pius** [comment]
Verifying on DEV with Kyma `9098f8ce-53e7-4358-a2c1-991dd90868dc`. We can see that the update is waiting:

![Image](https://github.com/user-attachments/assets/df3299e5-3a6a-425e-83a9-31fd8ce7f7a9)

According to policy, the maintenance window for this cluster is kicking in Saturday night.
